src/main/java/org/grouplens/grapht/reflect/internal/ClassInstantiator.java

```diff
@@ -74,7 +78,10 @@ public Class getType() {
     @Override
     public Object instantiate() throws ConstructionException {
         // find constructor and build up necessary constructor arguments
+
+        LogContext mdcContext = LogContext.create();
         Constructor<?> ctor = getConstructor();
+        mdcContext.put("org.grouplens.grapht.class", ctor.getClass().toString());
         Object[] ctorArgs = new Object[ctor.getParameterTypes().length];
         for (Desire d: desires) {
             if (d.getInjectionPoint() instanceof ConstructorParameterInjectionPoint) {
```

Comment:
I think we want to put a log context here too, to record the injection point in the key `org.grouplens.grapht.injectionPoint`. That will be useful for the code to see where the constructor is being injected.

Also, we should put in a `trace`-level message here that indicates the injection point we are trying to satisfy.

We should also do this below, when we are satisfying the method and field injection points.